rdr_service/genomic/genomic_job_controller.py

```diff
@@ -459,23 +473,58 @@ def gem_results_to_report_state(self):
 
         self.job_result = GenomicSubProcessResult.SUCCESS
 
-    def ingest_records_from_message_broker_data(self, *, message_record_id: int, event_type: str) -> None:
+    def reconcile_appointment_events_from_metrics(self):
+        appointment_metrics_dao = GenomicAppointmentEventMetricsDao()
+        metrics_missing_appointments = appointment_metrics_dao.get_missing_appointments()
 
-        module_fields = ['module_type', 'result_type']
+        if not metrics_missing_appointments:
+            logging.info('There are no missing appointment events to reconcile')
+            self.job_result = GenomicSubProcessResult.NO_RESULTS
+            return
 
-        def _set_value_from_parsed_values(
-            records: List[MessageBrokerEventData],
-            field_names: List[str]
-        ) -> Optional[str]:
+        logging.info(f'Creating {len(metrics_missing_appointments)} missing appointment events from metrics')
 
-            records = [records] if type(records) is not list else records
-            field_records = list(filter(lambda x: x.fieldName in field_names, records))
-            if not field_records:
-                return None
+        metrics_to_update = []
+        for metric in metrics_missing_appointments:
+            try:
+                if not metric.appointment_event_id:
+                    logging.info(
+                        f'Inserting appointment event from metric_id {metric.id} for participant'
+                        f' {metric.participant_id}'
+                    )
 
-            field_records = field_records[0].asdict()
-            value = [v for k, v in field_records.items() if v is not None and 'value' in k]
-            return value[0] if value else None
+                    appointment_obj = {
+                        'participant_id': metric.participant_id,
+                        'event_type': metric.event_type,
+                        'event_authored_time': metric.event_authored_time,
+                        'created_from_metric_id': metric.id
+                    }
+
+                    message_body = json.loads(metric.appointment_event).get('messageBody')
+                    for k, value in message_body.items():
+                        key = 'appointment_id' if k == 'id' else k
+                        appointment_obj[key] = value
+
+                    appointment_obj = self.appointment_dao.get_model_obj_from_items(appointment_obj.items())
+                    self.appointment_dao.insert(appointment_obj)
```

Comment:
You may have a reason for it, but since you are inserting the appointments here, do you need to update the reconciliation job run ID? In your query can you just look for total missing appointments instead of filtering on reconciliation ID? Once they are inserted they won't be returned from the missing query.

Reply:
Yea I was totally on the fence with that tbh. There really is no need for it I guess and looking at the logic for the other reconciles we arent using it. I was just wondering about how performant the query would be as we got so many of these events stored and figured this would be a way to mitigate the memory load caused by the query if I had another value to filter on.

Reply:
Sounds good, we'll monitor in case.